runtime/src/bank_forks.rs

```diff
@@ -290,7 +290,33 @@ impl BankForks {
         self.root
     }
 
-    fn prune_non_root(&mut self, root: Slot, highest_confirmed_root: Option<Slot>) {
+    /// After setting a new root, prune the banks that are no longer on rooted paths
+    ///
+    /// Given the following banks and slots...
+    ///
+    /// slot 6                   * (G)
+    ///                         /
+    /// slot 5        (F)  *   /
+    ///                    |  /
+    /// slot 4    (E) *    | /
+    ///               |    |/
+    /// slot 3        |    * (D) <-- root, from set_root()
+    ///               |    |
+    /// slot 2    (C) *    |
+    ///                \   |
+    /// slot 1          \  * (B)
+    ///                  \ |
+    /// slot 0             * (A)  <-- highest confirmed root
+    ///
+    /// ...where (D) is set as root, clean up (C) and (E), since they are not rooted.
+    /// (A) is kept because it is a lower slot than (D), and (D) is one of its decendants
```

Comment:
nit: decendants-> descendants

Reply:
nit: Also add because `A, B` are greater than `highest confirmed root`. We keep these around because RPC has a commitment level that is based on the cluster-confirmed root, which is a stronger commitment level than just a local root